include/roaring/roaring_types.h

```diff
@@ -97,6 +97,23 @@ typedef struct roaring_statistics_s {
     // and n_values_arrays, n_values_rle, n_values_bitmap
 } roaring_statistics_t;
 
+/**
+ * Roaring-internal type used to iterate within a roaring container.
+ *
+ * Only `value` and `has_value` should be accessed outside of containers, the
+ * rest of the type should be treated as opaque. `value` is only valid if
+ * `has_value` is true.
+ */
+typedef struct roaring_container_iterator_s {
+    uint16_t value;
+    bool has_value;
```

Comment:
In my opinion accessors would be an improvement because it allows us to hide the type and change it in the future without ABI incompatibility. It also makes the fields easier to access in other languages. However, it would be an API change.

Reply:
Hiding the type means requiring an allocation and requiring clients to call a cleanup function though, which can be unfortunate. Right now, there's no allocation, and once they're done iterating, they can just return or whatever.

I honestly think it would be a good idea to do it for `roaring_bitmap_t` (totally unrelated to your changes), but I'm less sold on the iterator.

That said, we _could_ still use accessors (inline?) even if we don't end up hiding the details of the type.

Reply:
Fully agreed on `roaring_bitmap_t`.

Regarding duplication, I've pushed a new commit that removes `value` and `has_value` in favor of out parameters. I think it's a bit better, though the function signature of `container_iterator_read_into_uint32` is not looking great.

Happy to add accessors to the roaring iterator, though as of right now it won't help reduce the size of the iterator.